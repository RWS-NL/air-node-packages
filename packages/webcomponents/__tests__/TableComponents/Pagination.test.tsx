import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import TablePagination from '../../src/TableComponents/Pagination/Pagination';

let tablePaginations: ShallowWrapper;
const mockOnChangePage = jest.fn();
const mockOnchangeRowsPerPage = jest.fn();

beforeAll(() => {
  tablePaginations = shallow(
    <TablePagination
      labelRowsPerPage='rows per page' labelPaginationOf='of'
      rowsPerPageOptions={[ 5, 10 ]} rowsPerPage={5}
      count={20} page={0}
      onChangePage={mockOnChangePage} onChangeRowsPerPage={mockOnchangeRowsPerPage}
    />);
});

test('should match snapshot', () => {
  expect(tablePaginations).toMatchSnapshot();
});

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const tablePagination = shallow(
      <TablePagination
        labelRowsPerPage='rows per page' labelPaginationOf='of'
        rowsPerPageOptions={[ 2, 5, 10 ]} rowsPerPage={5}
        count={20} page={0}
        onChangePage={mockOnChangePage} onChangeRowsPerPage={mockOnchangeRowsPerPage}
      />
    );
    expect(tablePagination).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const tablePagination = shallow(
      <TablePagination
        labelRowsPerPage='rows per page' labelPaginationOf='of'
        rowsPerPageOptions={[ 2, 5, 10 ]} rowsPerPage={5}
        count={20} page={0}
        onChangePage={mockOnChangePage} onChangeRowsPerPage={mockOnchangeRowsPerPage}
        data-qa='snapshot-qa' ActionsComponent='div'
        customClasses={[ 'snapshot-class' ]}
      />
    );
    expect(tablePagination).toMatchSnapshot();
  });
});