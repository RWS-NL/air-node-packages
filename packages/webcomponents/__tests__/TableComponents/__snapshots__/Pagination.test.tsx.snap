// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Optional Props 1`] = `
<WithStyles(ForwardRef(TablePagination))
  ActionsComponent={
    Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "type": [Function],
    }
  }
  className="tablePagination snapshot-class"
  classes={
    Object {
      "menuItem": "text",
      "root": "text",
      "selectIcon": "selectIcon",
    }
  }
  component="span"
  count={20}
  data-qa="snapshot-qa"
  labelDisplayedRows={[Function]}
  labelRowsPerPage="rows per page"
  onChangePage={[MockFunction]}
  onChangeRowsPerPage={[MockFunction]}
  page={0}
  rowsPerPage={5}
  rowsPerPageOptions={
    Array [
      2,
      5,
      10,
    ]
  }
/>
`;

exports[`Snapshot Testing Required Props 1`] = `
<WithStyles(ForwardRef(TablePagination))
  ActionsComponent={
    Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "type": [Function],
    }
  }
  className="tablePagination"
  classes={
    Object {
      "menuItem": "text",
      "root": "text",
      "selectIcon": "selectIcon",
    }
  }
  component="span"
  count={20}
  labelDisplayedRows={[Function]}
  labelRowsPerPage="rows per page"
  onChangePage={[MockFunction]}
  onChangeRowsPerPage={[MockFunction]}
  page={0}
  rowsPerPage={5}
  rowsPerPageOptions={
    Array [
      2,
      5,
      10,
    ]
  }
/>
`;

exports[`should match snapshot 1`] = `
<WithStyles(ForwardRef(TablePagination))
  ActionsComponent={
    Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "type": [Function],
    }
  }
  className="tablePagination"
  classes={
    Object {
      "menuItem": "text",
      "root": "text",
      "selectIcon": "selectIcon",
    }
  }
  component="span"
  count={20}
  labelDisplayedRows={[Function]}
  labelRowsPerPage="rows per page"
  onChangePage={[MockFunction]}
  onChangeRowsPerPage={[MockFunction]}
  page={0}
  rowsPerPage={5}
  rowsPerPageOptions={
    Array [
      5,
      10,
    ]
  }
/>
`;
